src/chat.rs

```diff
@@ -152,6 +180,123 @@ impl ChatId {
         self.set_blocked(context, Blocked::Not).await;
     }
 
+    /// Set protection without sending a message.
+    /// Used when a message arrives indicating that someone else has
+    /// changed the protection value for a chat.
+    pub(crate) async fn inner_set_protection(
+        self,
+        context: &Context,
+        protect: ProtectionStatus,
+    ) -> Result<(), Error> {
+        ensure!(!self.is_special(), "set protection: invalid chat-id.");
+
+        let chat = Chat::load_from_db(context, self).await?;
+
+        if protect == chat.protected {
+            info!(context, "Protection status unchanged for {}.", self);
+            return Ok(());
+        }
+
+        match protect {
+            ProtectionStatus::Protected => match chat.typ {
+                Chattype::Single | Chattype::Group => {
+                    let contact_ids = get_chat_contacts(context, self).await;
+                    for contact_id in contact_ids.into_iter() {
+                        let contact = Contact::get_by_id(context, contact_id).await?;
+                        if contact.is_verified(context).await != VerifiedStatus::BidirectVerified {
+                            bail!(
+                                "{} is not verified; cannot enable protection.",
+                                contact.get_display_name()
+                            );
```

Comment:
i think, this is fine. `inner_set_protection()` is called by `set_protection()` that calls `error!` if the Result-object indicates an error.

Reply:
Yes, my question here is that we only report the first unverified user.  I was wondering if someone tried to create a group with multiple un-verified contacts whether it made sense to show all the non-verified ones.  otherwise the user might end up fixing one only to find another that's problematic.  it's perhaps a bit far fetched, but also not that hard to do here, create a vector of unverified contacts and error with them all listed if it's non-zero

Reply:
i think, the ui will check is_verified() for each member before actually adding the member, so this error might not happen in the normal ui flow at all and is more a fallback if things went wrong for some reasons.

however, following https://github.com/orgs/deltachat/projects/28, adding switch-ui will be the second step, so if we need improvements, we can add that later as needed.